lib/postinstall.js

```diff
@@ -51,78 +58,67 @@ to work as it should:
   });
-}));
 
-const writeToPackageJson = (new Promise((resolve, reject) => {
-  const packageJsonPath = path.join(projectPath, 'package.json');
+  const packageJson = JSON.parse(content);
 
-  fs.readFile(packageJsonPath, 'utf-8', (err, content) => {
-    if (err) {
-      console.log('🤔  package.json not found, have you run `npm init`?');
-     return reject(err);
-    }
-
-    const packageJson = JSON.parse(content);
-
-    // Husky upgrade from 0.14.0 to ^1
-    // Remove old scripts.precommit
-    if ((packageJson.scripts || {}).precommit === 'lint-staged') {
-      delete packageJson.scripts.precommit;
-    }
-
-    // Set new husky hook
-    if (!packageJson.husky) {
-      packageJson.husky = {};
-    }
-    if (!packageJson.husky.hooks) {
-      packageJson.husky.hooks = {};
-    }
-    packageJson.husky.hooks['pre-commit'] = 'lint-staged';
-
-
-    // Default configuration for lint-staged
-    const lintStaged = {
-      '*.{js,json,graphql,md,css,scss,less,ts}': [
-        'prettier --write',
-        'git add'
-      ]
-    };
-
-    // Set or update lint-staged configuration
-    if (packageJson['lint-staged']) {
-      console.warn(`⚠️  A 'lint-staged' configuration already exists in package.json.
+  // Husky upgrade from 0.14.0 to ^1
+  // Remove old scripts.precommit
+  if ((packageJson.scripts || {}).precommit === 'lint-staged') {
+    delete packageJson.scripts.precommit;
+  }
+
+  // Set new husky hook
+  if (!packageJson.husky) {
+    packageJson.husky = {};
+  }
+  if (!packageJson.husky.hooks) {
+    packageJson.husky.hooks = {};
+  }
+  packageJson.husky.hooks['pre-commit'] = 'lint-staged';
+
+
+  // Default configuration for lint-staged
+  const lintStaged = {
+    '*.{js,json,graphql,md,css,scss,less,ts}': [
+      'prettier --write',
+      'git add'
+    ]
+  };
+
+  // Set or update lint-staged configuration
+  if (packageJson['lint-staged']) {
+    console.warn(`⚠️  A 'lint-staged' configuration already exists in package.json.
 We won't overwrite it since it may include some of your own customizations.
 We would have added the following rules, so check your configuration and modify it if needed:
 
 ${JSON.stringify(lintStaged, null, 2)}`);
-    } else {
-      packageJson['lint-staged'] = lintStaged;
-    }
-
-    // Stringify package.json and write to file
-    const jsonString = JSON.stringify(packageJson, null, 2);
-    fs.writeFile(packageJsonPath, jsonString, 'utf-8', (err) => {
-      if (err) {
-       return reject(err);
-      }
-
-     return resolve();
-    });
-  });
-}));
+  } else {
+    packageJson['lint-staged'] = lintStaged;
+  }
+
+  // Stringify package.json and write to file
+  const jsonString = JSON.stringify(packageJson, null, 2);
+
+  return writeFileAsync(packageJsonPath, jsonString, 'utf-8');
+};
 
-Promise.all([writeEslintRc, writePrettierRc, writeToPackageJson]).then(() => {
+Promise.all([writeEslintRc(), writePrettierRc(), writeToPackageJson()]).then(() => {
```

Comment:
Should `package.json` be done first? The others are bit moot if it doesn't exist

Reply:
Yes, I thought the same. Let's do it.